tools/md5_lookup.rb

```diff
@@ -0,0 +1,489 @@
+  $stdout.puts "[-] #{msg}"
+end
+
+module Md5LookupUtility
+
+  # This class manages the disclaimer
+  class Disclaimer
+
+    # @!attribute config_file
+    #  @return [String] The config file path
+    attr_accessor :config_file
+
+    # @!attribute group_name
+    #  @return [String] The name of the tool
+    attr_accessor :group_name
+
+
+    def initialize
+      self.config_file = Msf::Config.config_file
+      self.group_name  = 'MD5Lookup'
+    end
+
+    # Prompts a disclaimer. The user will not be able to get out unless they acknowledge.
+    #
+    # @return [TrueClass] true if acknowledged.
+    def ack
+      print_status("WARNING: This tool will look up your MD5 hashes by submitting them")
+      print_status("in the clear (HTTP) to third party websites. This can expose")
+      print_status("sensitive data to unknown and untrusted entities.")
+
+      while true
+        $stdout.print "[*] Enter 'Y' to acknowledge: "
+          if $stdin.gets =~ /^y|yes$/i
+            return true
+          end
+      end
+    end
+
+
+    # Saves the waiver so the warning won't show again after ack
+    #
+    # @return [void]
+    def save_waiver
+      save_setting('waiver', true)
+    end
+
+
+    # Returns true if we don't have to show the warning again
+    #
+    # @return [Boolean]
+    def has_waiver?
+      load_setting('waiver') == 'true' ? true : false
+    end
+
+
+    private
+
+    # Saves a setting to Metasploit's config file
+    #
+    # @param key_name [String] The name of the setting
+    # @param value [String] The value of the setting
+    # @return [void]
+    def save_setting(key_name, value)
+      ini = Rex::Parser::Ini.new(self.config_file)
+      ini.add_group(self.group_name) if ini[self.group_name].nil?
+      ini[self.group_name][key_name] = value
+      ini.to_file(self.config_file)
+    end
+
+
+    # Returns the value of a specific setting
+    #
+    # @param key_name [String] The name of the setting
+    # @return [String]
+    def load_setting(key_name)
+      ini = Rex::Parser::Ini.new(self.config_file)
+      group = ini[self.group_name]
+      return '' if group.nil?
+      group[key_name].to_s
+    end
+
+  end
+
+  # This class is basically an auxiliary module without relying on msfconsole
+  class Md5Lookup < Msf::Auxiliary
+
+    include Msf::Exploit::Remote::HttpClient
+
+    # @!attribute rhost
+    #  @return [String] Should be md5cracker.org
+    attr_accessor :rhost
+
+    # @!attribute rport
+    #  @return [Fixnum] The port number to md5cracker.org
+    attr_accessor :rport
+
+    # @!attribute target_uri
+    #  @return [String] The URI (API)
+    attr_accessor :target_uri
+
+    # @!attribute ssl
+    #  @return [FalseClass] False because doesn't look like md5cracker.org supports HTTPS
+    attr_accessor :ssl
+
+    def initialize(opts={})
+      # The user should not be able to modify these settings, otherwise
+      # the we can't guarantee results.
+      self.rhost      = 'md5cracker.org'
```

Comment:
Maybe should be constants? Not an stopper to me, just curiosity! I won't hold up landing!